implementations/rust/ockam/ockam_command/src/util/parsers.rs

```diff
@@ -21,6 +21,17 @@ pub(crate) fn hostname_parser(input: &str) -> Result<SchemeHostnamePort> {
     ))
 }
 
+/// Helper function for parsing a key-value pair from user input
+/// The input is expected to be in the format `key:value`
+pub(crate) fn http_header_parser(input: &str) -> Result<(String, String)> {
+    let parts: Vec<&str> = input.split(':').collect();
+    if parts.len() != 2 {
+        return Err(miette!("Invalid header format. Expected 'key:value'"));
+    }
+
+    Ok((parts[0].trim().to_string(), parts[1].trim().to_string()))
```

Comment:
Ideally we could check [the specification for header names and values](https://datatracker.ietf.org/doc/html/rfc7230#page-22), which is like:

- The header name has one or more characters in this range:
`! # $ % & ' * + - .  ^ _ ` | ~ digit alpha`
```
- The value is non-empty, and does not contain newlines (`\n`, `\r\n`) or control characters (`0x00–0x1F, 0x7F`).

This way we can prevent hard-to-diagnose issues down the stream. One typical example would be to define:
```
--header token:$AUTH_TOKEN
```
Where the user makes a mistake and uses `AUTH_TOKEN` instead of `AUTHORIZATION_TOKEN`.

Reply:
I've taken inspiration from the `httparse` library we're using and made the parsing much more robust!
And I've disallowed an empty value, so we can cover the empty variable resolution case.